src/ui/components/uicontroller/UiContainer.tsx

```diff
@@ -341,36 +398,50 @@ export class UiContainer extends PureComponent<React.PropsWithChildren<UiContain
+        <View style={FULLSCREEN_CENTER_STYLE} pointerEvents={'none'}>
+          {behind}
+        </View>
         {/* The Animated.View is for showing and hiding the UI*/}
-        <Animated.View
-          style={[combinedContainerStyle, { opacity: fadeAnimation }]}
-          onTouchStart={this.onUserAction_}
-          pointerEvents={showing ? 'auto' : 'box-only'}
-          {...(Platform.OS === 'web' ? { onMouseMove: this.onUserAction_, onMouseLeave: this.doFadeOut_ } : {})}>
-          <>
-            {/* The UI background */}
-            <View style={[combinedContainerStyle, { backgroundColor: theme.colors.uiBackground }]} onTouchStart={this.doFadeOut_} />
-
-            {/* The Settings Menu */}
-            {currentMenu !== undefined && <View style={[combinedContainerStyle]}>{currentMenu}</View>}
-
-            {/* The UI control bars*/}
-            {currentMenu === undefined && (
-              <>
-                {firstPlay && <View style={[TOP_UI_CONTAINER_STYLE, topStyle]}>{top}</View>}
-                <View style={[CENTER_UI_CONTAINER_STYLE, centerStyle]}>{center}</View>
-                {firstPlay && <View style={[BOTTOM_UI_CONTAINER_STYLE, bottomStyle]}>{bottom}</View>}
-                {children}
-              </>
-            )}
-          </>
-        </Animated.View>
+        {!showMobileAdLayout && (
+          <Animated.View
+            style={[combinedUiContainerStyle, { opacity: fadeAnimation }]}
+            onTouchStart={this.onUserAction_}
+            pointerEvents={showing ? 'auto' : 'box-only'}
+            {...(Platform.OS === 'web' ? { onMouseMove: this.onUserAction_, onMouseLeave: this.doFadeOut_ } : {})}>
+            <>
+              {/* The UI background */}
+              <View style={[combinedUiContainerStyle, { backgroundColor: theme.colors.uiBackground }]} onTouchStart={this.doFadeOut_} />
+
+              {/* The Settings Menu */}
+              {currentMenu !== undefined && <View style={[combinedUiContainerStyle]}>{currentMenu}</View>}
+
+              {/* The UI control bars*/}
+              {currentMenu === undefined && (
+                <>
+                  {firstPlay && <View style={[TOP_UI_CONTAINER_STYLE, topStyle]}>{top}</View>}
+                  <View style={[CENTER_UI_CONTAINER_STYLE, centerStyle]}>{center}</View>
+                  {!adInProgress && firstPlay && <View style={[BOTTOM_UI_CONTAINER_STYLE, bottomStyle]}>{bottom}</View>}
+                  {adInProgress && <View style={[AD_UI_CONTAINER_STYLE, adStyle]}>{ad}</View>}
+                  {children}
+                </>
+              )}
+            </>
+          </Animated.View>
+        )}
+        {/* Simplistic ad view to allow play pause during an ad on mobile. */}
+        {showMobileAdLayout && (
+          <View style={[combinedAdContainerStyle]}>
+            <TouchableOpacity style={[FULLSCREEN_CENTER_STYLE]} onPress={this.playPause_}></TouchableOpacity>
+          </View>
+        )}
```

Comment:
On mobile, tapping the player usually doesn't toggle its paused state. Would users find this behavior surprising?

Reply:
Yea I agree it isn't ideal at the moment. We discussed that initially we deliver minimal ad support with this and later on improve the layout once we figure out how to properly work with the pointer/touch events.